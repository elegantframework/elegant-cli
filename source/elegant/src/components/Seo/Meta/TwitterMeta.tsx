interface Props {
    handle?: string;
    site?: string;
};

/**
 * Twitter card meta data.
 * @returns A list of twitter card meta data items for the seo component.
 */
const TwitterMeta = ({
    handle = "",
    site = "@"
}: Props) => {
    if(handle !== ""){
        // if a site twitter account has been provided
        if(site !== "@"){
            return {
                handle: handle,
                site: site,
                cardType: 'summary_large_image',
            };
        }
        
        // no site provided.. Render only a twitter handle and card
        return {
            handle: handle,
            cardType: 'summary_large_image',   
        };
    }
    else{
        return {
            cardType: 'summary_large_image',   
        };
    }
};

export default TwitterMeta;