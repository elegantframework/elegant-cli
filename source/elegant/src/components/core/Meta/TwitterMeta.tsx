
interface Props {
    /**
     * The website creators Twitter handle. Ex. @tailwindcss
     */
    twitterHandle: string;
};

/**
 * If a twitter handle is provided, return the twitter heading card meta data.
 * @param param0 component props.
 * @returns Twitter site and creator meta heading tags.
 */
const TwitterMeta = ({twitterHandle}: Props) => 
{
    // if there isn't a handle provided, return nothing
    if(twitterHandle === "")
    {
        return null;
    }

    return(
        <>
            <meta key="twitter:site" name="twitter:site" content="@thebrandonowens" />
            <meta key="twitter:creator" name="twitter:creator" content="@thebrandonowens" />
        </> 
    );
}

export default TwitterMeta;