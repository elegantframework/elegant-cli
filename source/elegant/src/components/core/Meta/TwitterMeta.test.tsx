import React from "react";
import renderer from 'react-test-renderer';
import TwitterMeta from "./TwitterMeta";

describe("Twitter meta card component", () => {
    it('renders a twitter creator card if a creator is provided',() => {
        const card = renderer
            .create(<TwitterMeta twitterHandle="@test"/>)
            .toJSON();
        expect(card).toMatchSnapshot();
    });

    it('renders nothing if a handle is not provided',() => {
        const card = renderer
            .create(<TwitterMeta twitterHandle=""/>)
            .toJSON();
        expect(card).toBeNull();
        expect(card).toMatchSnapshot();
    });
});