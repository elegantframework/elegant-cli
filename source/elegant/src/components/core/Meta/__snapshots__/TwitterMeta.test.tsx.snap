// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter meta card component renders a twitter creator card if a creator is provided 1`] = `
[
  <meta
    content="@thebrandonowens"
    name="twitter:site"
  />,
  <meta
    content="@thebrandonowens"
    name="twitter:creator"
  />,
]
`;

exports[`Twitter meta card component renders nothing if a handle is not provided 1`] = `null`;
